geode-core/src/main/java/org/apache/geode/internal/cache/wan/parallel/ParallelGatewaySenderQueue.java

```diff
@@ -1781,4 +1785,47 @@ ParallelGatewaySenderQueueMetaRegion newMetataRegion(InternalCache cache, final
       return meta;
     }
   }
+
+
+
+  public void clearQueueTestOnly() {
+
+    this.sender.pause();// it wil take internal read-write-lock
+    try {
+      for (PartitionedRegion prQ : this.userRegionNameToshadowPRMap.values()) {
+        clearPartitionedRegionTestOnly((PartitionedRegion) prQ);
+      }
+    } catch (Exception ee) {
+      throw ee;
+    } finally {
+      if (this.sender.isPaused())
+        this.sender.resume();
+    }
+  }
+
+  // clear the partition region
+  private void clearPartitionedRegionTestOnly(PartitionedRegion partitionedRegion) {
+    LocalDataSet lds = (LocalDataSet) PartitionRegionHelper.getLocalPrimaryData(partitionedRegion);
+    Set<Integer> set = lds.getBucketSet(); // this returns bucket ids in the
+                                           // function context
+
+    for (Integer bucketId : set) {
+      Bucket bucket = partitionedRegion.getRegionAdvisor().getBucket(bucketId);
+      if (bucket instanceof ProxyBucketRegion == false) {
+        if (bucket instanceof BucketRegion) {
+          BucketRegion bucketRegion = (BucketRegion) bucket;
+          Set keySet = bucketRegion.keySet();
+          for (Iterator iterator = keySet.iterator(); iterator.hasNext();) {
+            Object key = iterator.next();
+            try {
+              bucketRegion.remove(key);
+            } catch (Exception ee) {
+              ee.printStackTrace(System.out);
```

Comment:
Remove all printStackTrace calls. If the exception is simply rethrown then remove the catch-block altogether.

Reply:
done